### PR TITLE
Cleanup misc-tools scripts

### PR DESCRIPTION
**Pull request by edomora97**
This PR tries to cleanup a bit the scripts inside misc-tools/. Since many lines are involved in the process, I've tried to split the PR in multiple commits for an easier review. They all try to "clean up the scripts", so in the end they can be squashed into a single commit if you want.

The idea of this PR is to clean up without making changes to the overall behaviour of the scripts, so I hope to not have introduced any side effect. Overall this PR does:

- uniform the indentation of the various scripts (4 spaces + 2 spaces for alignment)
- fix SC2086 (missing quotes) ¹
- fix other shellcheck warnings ²
- try to reduce code duplication by extracting some variables and functions (most notably `in_chroot` in the last commit)

¹ I'm aware of [this script](https://github.com/DOMjudge/domjudge/blob/main/tests/syntax#L52) that should run shellcheck on the scripts, and I see why SC2086 is ignored by default. Sill, I think that, to be on the safe side, the quotes should be added.

² I don't know why many of the warnings I've fixed were not found by that script in the CI (is it included in the pipeline?). In any case I used `shellcheck 0.7.2`.

**Disclaimer**: I've tried to test the changes as much as possible, but I'm not able to run few of the scripts: `compare-cds.sh`, `dj_make_chroot_docker` (~doesn't work yet on my machine, maybe a new PR will arrive 👀~, I've fixed it, a new PR will arrive soon since it's unrelated to this), `fix_permissions`, `import-contest.sh`, `jvm_footprint`.

### CODE REVIEW

**Comment by edomora97** [comment]
Sorry, I've to force push since I forgot to [indent a section](https://github.com/DOMjudge/domjudge/compare/d16b186828b1238c99d5b9530e075a59a3c2705d..7ebded63b69e165214bdc708fbf06aea91290eab) in `dj_make_chroot`.

**Review by nickygerritsen — APPROVED** [review]

File: misc-tools/dj_make_chroot.in [hunk 1]
```diff
@@ -336,7 +336,11 @@ mount --bind /dev/pts "$CHROOTDIR/dev/pts"
 # Prevent perl locale warnings in the chroot:
 export LC_ALL=C
 
-chroot "$CHROOTDIR" /bin/sh -c debconf-set-selections <<EOF
+in_chroot() {
+    chroot "$CHROOTDIR" "$@"
```
Comment (eldering) [inline misc-tools/dj_make_chroot.in:340]: I think it makes sense to also include `/bin/sh -c` here as all calls of `in_chroot` use that.

File: misc-tools/dj_judgehost_cleanup.in [hunk 2]
```diff
@@ -83,10 +83,9 @@ case $1 in
             [ -d "$d" ] || continue
             cd "$d"
             echo "  in $d:"
-            for i in $(ls -d * 2>/dev/null | grep -E '^[0-9]+$' || true); do
-                [ -d "$i" ] || continue
-                echo "    removing judging data for contest $i"
-                rm -rf "$i/"
+            find . -type d -regex '\./[0-9]+' | while IFS= read -r contest; do
```
Comment (eldering) [inline misc-tools/dj_judgehost_cleanup.in:86]: Nitpick: add `-maxdepth 1` to not traverse deeper than needed.

File: misc-tools/dj_make_chroot.in [hunk 3]
```diff
@@ -35,15 +35,15 @@ DEBIAN_ARCHLIST="amd64,arm64,armel,armhf,i386,mips,mips64el,mipsel,ppc64el,s390x
 UBUNTU_ARCHLIST="amd64,arm64,armhf,i386,powerpc,ppc64el,s390x"
 
 # If host system is Debian or Ubuntu, use its release and architecture by default
-if [ "x$(lsb_release -i -s || true)" = 'xDebian' ] || \
-   [ "x$(lsb_release -i -s || true)" = 'xUbuntu' ]; then
+if [ "$(lsb_release -i -s || true)" = 'Debian' ] || \
+   [ "$(lsb_release -i -s || true)" = 'Ubuntu' ]; then
```
Comment (eldering) [inline misc-tools/dj_make_chroot.in:39]: Was shellcheck complaining about this? I realize this is maybe (over)cautious escaping, but doesn't hurt as far as I'm aware.

File: misc-tools/dj_make_chroot.in [hunk 4]
```diff
@@ -384,17 +384,17 @@ chroot "$CHROOTDIR" /bin/sh -c "apt-get clean"
...
+  /bin/su /bin/mount /bin/umount /sbin/unix_chkpwd" || true
 
 # Disable root account
 sed -i "s/^root::/root:*:/" "$CHROOTDIR/etc/shadow"
 
 # Create a file to test that in the judging environment no
 # access to group root readable files is available:
 echo "This file should not be readable inside the judging environment!" \
-	> "$CHROOTDIR/etc/root-permission-test.txt"
+  > "$CHROOTDIR/etc/root-permission-test.txt"
```
Comment (eldering) [inline misc-tools/dj_make_chroot.in:401]: This indentation looks off.

File: misc-tools/dj_make_chroot.in [hunk 5]
```diff
@@ -384,17 +384,17 @@ chroot "$CHROOTDIR" /bin/sh -c "apt-get clean"
 
 # Remove unnecessary setuid bits
 chroot "$CHROOTDIR" /bin/sh -c "chmod a-s /usr/bin/wall /usr/bin/newgrp \
-	/usr/bin/chage /usr/bin/chfn /usr/bin/chsh /usr/bin/expiry \
-	/usr/bin/gpasswd /usr/bin/passwd \
-	/bin/su /bin/mount /bin/umount /sbin/unix_chkpwd" || true
+  /usr/bin/chage /usr/bin/chfn /usr/bin/chsh /usr/bin/expiry \
+  /usr/bin/gpasswd /usr/bin/passwd \
+  /bin/su /bin/mount /bin/umount /sbin/unix_chkpwd" || true
```
Comment (eldering) [inline misc-tools/dj_make_chroot.in:393]: This indentation looks off.

File: misc-tools/dj_make_chroot_docker.in [hunk 6]
```diff
@@ -142,7 +142,7 @@ sed -i "s/^root::/root:*:/" "$CHROOTDIR/etc/shadow"
 # Create a file to test that in the judging environment no
 # access to group root readable files is available:
 echo "This file should not be readable inside the judging environment!" \
-	> "$CHROOTDIR/etc/root-permission-test.txt"
+  > "$CHROOTDIR/etc/root-permission-test.txt"
```
Comment (eldering) [inline misc-tools/dj_make_chroot_docker.in:146]: This indentation looks off.

File: misc-tools/dj_make_chroot.in (hunk 3 above)
Comment (vmcj) [reply misc-tools/dj_make_chroot.in:39]: I think the removal of `x` is because of: https://github-wiki-see.page/m/koalaman/shellcheck/wiki/SC2268

Comment (edomora97) [reply misc-tools/dj_make_chroot.in:39]: Yes, it was exactly that lint.

File: misc-tools/dj_make_chroot.in (hunk 4 above)
Comment (edomora97) [reply misc-tools/dj_make_chroot.in:401]: The rationale was to indent using 4 spaces, but use 2 for multi-line commands. If you prefer I'll use 4 spaces everywhere.

**Comment by vmcj** [comment]
@edomora97 is this ready for review again?

**Comment by edomora97** [comment]
> @edomora97 is this ready for review again?

Almost, there is still the question about the [indentation level for the line breaks](https://github.com/DOMjudge/domjudge/pull/1393#discussion_r773500057) pending. Do you have any opinion about that?

**Comment by vmcj** [comment]
> > @edomora97 is this ready for review again?
> 
> Almost, there is still the question about the [indentation level for the line breaks](https://github.com/DOMjudge/domjudge/pull/1393#discussion_r773500057) pending. Do you have any opinion about that?

I think for simplicity we should go for 4 spaces,

**Comment by vmcj** [comment]
@edomora97 thanks for the great work, I'll merge it.

**Pull request merged**